specs/decimal/decimal.md

```diff
@@ -68,125 +229,106 @@ namespace Microsoft.Windows.Foundation
-        String ToStringWithInvariantLocale();               // LCID=LOCALE_INVARIANT
-        IInspectable ToObject();
-        Decimal ToDecimal();                //TODO: Rename to Copy(value) or Clone(value) ?
-        DecimalValue ToDecimalValue();
-
-        /// Return true if (this == value).
-        Boolean Equals(Decimal value);
-
-        /// Compare this decimal with value.
-        /// @return 0 if this and value are equal, <0 if this is less than value or >0 if this is greater than value.
-        Int32 Compare(Decimal value);
+        static DecimalValue FromBoolean(Boolean value);
+        static DecimalValue FromInt16(Int16 value);
+        static DecimalValue FromInt32(Int32 value);
+        static DecimalValue FromInt64(Int64 value);
+        static DecimalValue FromUInt8(UInt8 value);
+        static DecimalValue FromUInt16(UInt16 value);
+        static DecimalValue FromUInt32(UInt32 value);
+        static DecimalValue FromUInt64(UInt64 value);
+        static DecimalValue FromSingle(Single value);
+        static DecimalValue FromDouble(Double value);
+        static DecimalValue FromString(String value);                          // LCID=LOCALE_INVARIANT
+        static DecimalValue FromStringWithSystemDefaultLocale(String value);   // LCID=GetSystemDefaultLCID()
+        static DecimalValue FromStringWithUserDefaultLocale(String value);     // LCID=GetUserDefaultLCID()
+        static DecimalValue FromStringWithThreadLocale(String value);          // LCID=GetThreadLocale()
+
+        static Boolean ToBoolean(DecimalValue value);
+        static Int16 ToInt16(DecimalValue value);
+        static Int32 ToInt32(DecimalValue value);
+        static Int64 ToInt64(DecimalValue value);
+        static UInt8 ToUInt8(DecimalValue value);
+        static UInt16 ToUInt16(DecimalValue value);
+        static UInt32 ToUInt32(DecimalValue value);
+        static UInt64 ToUInt64(DecimalValue value);
+        static Single ToSingle(DecimalValue value);
+        static Double ToDouble(DecimalValue value);
+        static String ToString(DecimalValue value);                         // LCID=LOCALE_INVARIANT
+        static String ToStringWithSystemDefaultLocale(DecimalValue value);  // LCID=GetSystemDefaultLCID()
+        static String ToStringWithUserDefaultLocale(DecimalValue value);    // LCID=GetUserDefaultLCID()
+        static String ToStringWithThreadLocale(DecimalValue value);         // LCID=GetThreadLocale()
+
+        /// Return true if (left == right).
+        static Boolean Equals(DecimalValue left, DecimalValue right);
+
+        /// Compare decimal values.
+        /// @return 0 if left and right are equal, <0 if left is less than right or >0 if left is greater than right.
+        static Int32 Compare(DecimalValue left, DecimalValue right);
+
+        /// Return true if the value is valid.
+        static Boolean IsValid(DecimalValue value);
```

Comment:
What does "valid" mean here?

Reply:
`DecimalValue` defines `Scale` as a byte/uint8 but only valid values are 0-28, and `Sign` is byte/uint8 but only valid values are 0 and 0x80.

Interestingly, at lest some of the `Var*Dec*()` functions detect this and error if invalid, and some don't. We'll be more rigorous in checking our inputs of course :-)